willisapi_client/services/metadata/archive.py

```
@@ -0,0 +1,83 @@
+            logger.warning(f"CSV archive init failed: {res.status_code} {res.text}")
+            return None
+
+        body = res.json()
+        record_id = body.get("record_id")
+        presigned = body.get("presigned_url")
+        if not presigned:
+            logger.warning("CSV archive: no presigned URL returned")
+            return None
+
+        with open(csv_path, "rb") as f:
+            put_res = requests.put(
```

Ruff (0.15.15)

[error] Probable use of `requests` call without timeout

(S113)